[YW] Making the registry of various images configurable

iSignal opened the pull request:
Summary: 
When YW is deployed in an environment with no internet access that uses a private image repo, we need to make all the different components of YW be pull-able from the private repo.

1. Each image can have a configurable registry and tag. I left the yugaware image configuration with the old process of configuring the repository tag directly.
2. A global override for all registries is also available and takes highest precedence for all images.

Test Plan:

Test with no overrides
```
sanketh@Sankeths-MacBook-Pro ~/code/charts repoconfig2$ helm template ~/code/charts/stable/yugaware > /tmp/tmp-charts/new.yaml; diff /tmp/tmp-charts/old.yaml /tmp/tmp-charts/new.yaml | egrep -v 'postgres_pass|app_sec'
35c35
---
37c37
---
578c578
<           image: "janeczku/go-dnsmasq:release-1.0.7"
---
>           image: janeczku/go-dnsmasq:release-1.0.7
```

Test with single registry override
```
anketh@Sankeths-MacBook-Pro ~/code/charts repoconfig2$ helm template ~/code/charts/stable/yugaware --set image.postgres.registry=quay1.io > /tmp/tmp-charts/new.yaml; diff /tmp/tmp-charts/old.yaml /tmp/tmp-charts/new.yaml | egrep -v 'postgres_pass|app_sec'
35c35
---
37c37
---
466c466
<         - image: postgres:11.5
---
>         - image: quay1.io/postgres:11.5
578c578
<           image: "janeczku/go-dnsmasq:release-1.0.7"
---
>           image: janeczku/go-dnsmasq:release-1.0.7
```

Test with commonRegistry override

```
sanketh@Sankeths-MacBook-Pro ~/code/charts repoconfig2$ helm template ~/code/charts/stable/yugaware --set image.commonRegistry=quay1.io > /tmp/tmp-charts/new.yaml; diff /tmp/tmp-charts/old.yaml /tmp/tmp-charts/new.yaml | egrep -v 'postgres_pass|app_sec'
35c35
---
37c37
---
466c466
<         - image: postgres:11.5
---
>         - image: quay1.io/postgres:11.5
494c494
<           image: prom/prometheus:v2.2.1
---
>           image: quay1.io/prom/prometheus:v2.2.1
511c511
<           image: quay.io/yugabyte/thirdparty-deps:latest
---
>           image: quay1.io/yugabyte/thirdparty-deps:latest
526c526
<           image: quay.io/yugabyte/yugaware:2.3.2.0-b37
---
>           image: quay1.io/yugabyte/yugaware:2.3.2.0-b37
571c571
<           image: nginx:1.17.4
---
>           image: quay1.io/nginx:1.17.4
578c578
<           image: "janeczku/go-dnsmasq:release-1.0.7"
---
>           image: quay1.io/janeczku/go-dnsmasq:release-1.0.7
```

Test with commmonRegistry override but also a specific tag and registry override. We expect commonRegistry to be reflected and the tag to be reflected.
```
sanketh@Sankeths-MacBook-Pro ~/code/charts repoconfig2$ helm template ~/code/charts/stable/yugaware --set image.commonRegistry=quay1.io,image.postgres.registry=quaypostgres.io,image.postgres.tag=9.0 > /tmp/tmp-charts/new.yaml; diff /tmp/tmp-charts/old.yaml /tmp/tmp-charts/new.yaml | egrep -v 'postgres_pass|app_sec'
35c35
---
37c37
---
466c466
<         - image: postgres:11.5
---
>         - image: quay1.io/postgres:9.0
494c494
<           image: prom/prometheus:v2.2.1
---
>           image: quay1.io/prom/prometheus:v2.2.1
511c511
<           image: quay.io/yugabyte/thirdparty-deps:latest
---
>           image: quay1.io/yugabyte/thirdparty-deps:latest
526c526
<           image: quay.io/yugabyte/yugaware:2.3.2.0-b37
---
>           image: quay1.io/yugabyte/yugaware:2.3.2.0-b37
571c571
<           image: nginx:1.17.4
---
>           image: quay1.io/nginx:1.17.4
578c578
<           image: "janeczku/go-dnsmasq:release-1.0.7"
---
>           image: quay1.io/janeczku/go-dnsmasq:release-1.0.7
```

bmatican commented on stable/yugabyte/Chart.yaml:
```diff
@@ -1,7 +1,7 @@
-version: 2.0.9
-appVersion: 2.0.9.0-b13
+version: 2.0.11
+appVersion: 2.0.11.0-b23
```
does this version have the percentile changes in YB?

bmatican commented on stable/yugaware/templates/service.yaml:
```diff
@@ -111,10 +109,12 @@ spec:
               subPath: postgres_data
         - name: prometheus
           image: prom/prometheus:v2.2.1
+          securityContext:
```
what's this about?

ramkumarvs replied:
@Arnav15 are we missing some more info here?

bmatican commented on stable/yugaware/templates/service.yaml:
```diff
@@ -111,10 +109,12 @@ spec:
           - name: prometheus-config
             mountPath: /etc/prometheus/
-          - name: prometheus-storage
+          - name: yugaware-storage
```
@ramkumarvs should prolly eyeball this

ramkumarvs replied:
LG. we chatted about this.

ajcaldera1 reviewed (approved):
LGTM. Let's get the same changes applied to yugabytedb chart as well.

iSignal committed:
[#1634] [YW] Making the registry of various images configurable

Summary:
When YW is deployed in an environment with no internet access that uses
a private image repo, we need to make all the different components of YW
be pull-able from the private repo.

1. Each image can have a configurable registry and tag. I left the
yugaware image configuration with the old process of configuring the
repository tag directly.
2. A global override for all registries is also available and takes
highest precedence for all images.

Test Plan:

Test with no overrides
```
sanketh@Sankeths-MacBook-Pro ~/code/charts repoconfig2$ helm template
~/code/charts/stable/yugaware > /tmp/tmp-charts/new.yaml; diff
/tmp/tmp-charts/old.yaml /tmp/tmp-charts/new.yaml | egrep -v
'postgres_pass|app_sec'
35c35
---
37c37
---
578c578
<           image: "janeczku/go-dnsmasq:release-1.0.7"
---
>           image: janeczku/go-dnsmasq:release-1.0.7
```

Test with single registry override
```
anketh@Sankeths-MacBook-Pro ~/code/charts repoconfig2$ helm template
~/code/charts/stable/yugaware --set image.postgres.registry=quay1.io >
/tmp/tmp-charts/new.yaml; diff /tmp/tmp-charts/old.yaml
/tmp/tmp-charts/new.yaml | egrep -v 'postgres_pass|app_sec'
35c35
---
37c37
---
466c466
<         - image: postgres:11.5
---
>         - image: quay1.io/postgres:11.5
578c578
<           image: "janeczku/go-dnsmasq:release-1.0.7"
---
>           image: janeczku/go-dnsmasq:release-1.0.7
```

Test with commonRegistry override

```
sanketh@Sankeths-MacBook-Pro ~/code/charts repoconfig2$ helm template
~/code/charts/stable/yugaware --set image.commonRegistry=quay1.io >
/tmp/tmp-charts/new.yaml; diff /tmp/tmp-charts/old.yaml
/tmp/tmp-charts/new.yaml | egrep -v 'postgres_pass|app_sec'
35c35
---
37c37
---
466c466
<         - image: postgres:11.5
---
>         - image: quay1.io/postgres:11.5
494c494
<           image: prom/prometheus:v2.2.1
---
>           image: quay1.io/prom/prometheus:v2.2.1
511c511
<           image: quay.io/yugabyte/thirdparty-deps:latest
---
>           image: quay1.io/yugabyte/thirdparty-deps:latest
526c526
<           image: quay.io/yugabyte/yugaware:2.3.2.0-b37
---
>           image: quay1.io/yugabyte/yugaware:2.3.2.0-b37
571c571
<           image: nginx:1.17.4
---
>           image: quay1.io/nginx:1.17.4
578c578
<           image: "janeczku/go-dnsmasq:release-1.0.7"
---
>           image: quay1.io/janeczku/go-dnsmasq:release-1.0.7
```

Test with commmonRegistry override but also a specific tag and registry
override. We expect commonRegistry to be reflected and the tag to be
reflected.
```
sanketh@Sankeths-MacBook-Pro ~/code/charts repoconfig2$ helm template
~/code/charts/stable/yugaware --set
image.commonRegistry=quay1.io,image.postgres.registry=quaypostgres.io,image.postgres.tag=9.0
> /tmp/tmp-charts/new.yaml; diff /tmp/tmp-charts/old.yaml
> /tmp/tmp-charts/new.yaml | egrep -v 'postgres_pass|app_sec'
35c35
---
37c37
---
466c466
<         - image: postgres:11.5
---
>         - image: quay1.io/postgres:9.0
494c494
<           image: prom/prometheus:v2.2.1
---
>           image: quay1.io/prom/prometheus:v2.2.1
511c511
<           image: quay.io/yugabyte/thirdparty-deps:latest
---
>           image: quay1.io/yugabyte/thirdparty-deps:latest
526c526
<           image: quay.io/yugabyte/yugaware:2.3.2.0-b37
---
>           image: quay1.io/yugabyte/yugaware:2.3.2.0-b37
571c571
<           image: nginx:1.17.4
---
>           image: quay1.io/nginx:1.17.4
578c578
<           image: "janeczku/go-dnsmasq:release-1.0.7"
---
>           image: quay1.io/janeczku/go-dnsmasq:release-1.0.7
```
